code/modules/mob/living/carbon/carbon.dm

```diff
@@ -982,12 +982,24 @@
 			if(ARM_LEFT)
 				l_arm_index_next += 2
 				bodypart_instance.held_index = l_arm_index_next //1, 3, 5, 7...
-				hand_bodyparts += bodypart_instance
+				on_added_hand(bodypart_instance, l_arm_index_next)
 			if(ARM_RIGHT)
 				r_arm_index_next += 2
 				bodypart_instance.held_index = r_arm_index_next //2, 4, 6, 8...
-				hand_bodyparts += bodypart_instance
+				on_added_hand(bodypart_instance, r_arm_index_next)
 
+/// Called when a new hand is added
+/mob/living/carbon/proc/on_added_hand(obj/item/bodypart/arm/new_hand, hand_index)
+	if(hand_index > hand_bodyparts.len)
+		hand_bodyparts.len = hand_index
+	hand_bodyparts[hand_index] = new_hand
```

Comment:
Do we loop over `hand_bodyparts` at any point? Will having nulls in there break things?

Reply:
Having nulls is the expected behaviour. The index is frequently used directly as a lookup into this list (potentially a bad idea begging for runtimes) so removing things from it _without_ leaving a null would runtime, and also change the order of the list.

Reply:
Okay, just making sure